components/contribute-cards/ContributeEventsPanel.js

```diff
@@ -0,0 +1,157 @@
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { Box } from '@rebass/grid';
+import dynamic from 'next/dynamic';
+
+import { Pencil } from 'styled-icons/boxicons-solid/Pencil';
+import { Move } from 'styled-icons/boxicons-regular/Move';
+import { Close } from 'styled-icons/material/Close';
+import styled from 'styled-components';
+import memoizeOne from 'memoize-one';
+import { get, set } from 'lodash';
+
+import CreateNew from './CreateNew';
+import ContributeEvent from './ContributeEvent';
+import ContributeCollective from './ContributeCollective';
+import { CONTRIBUTE_CARD_WIDTH } from './Contribute';
+import LoadingPlaceholder from '../LoadingPlaceholder';
+import colors from '../../lib/constants/colors';
+
+const Handle = styled.div`
+  background-color: ${({ bgColor }) => bgColor || colors.white};
+  box-shadow: 1px 1px 1px ${colors.darkgray};
+  opacity: ${({ hide }) => (hide ? 0 : 1)};
+  transition: all 0.3s ease-out;
+  justify-content: center;
+  color: ${colors.white};
+  align-items: center;
+  margin-bottom: 5px;
+  border-radius: 50%;
+  outline: none;
+  display: flex;
+  height: 30px;
+  width: 30px;
+  > * {
+    color: ${({ color }) => color || colors.darkgray};
+  }
+`;
+
+const ContributeEventsPanel = ({
+  isAdmin,
+  collective,
+  joinedEvents,
+  handleSettingsUpdate,
+  CONTRIBUTE_CARD_PADDING_X,
+  hasNoContributorForEvents,
+}) => {
+  const CONTRIBUTE_CARD_HEIGHT = 341;
+
+  const StyledDragDropPlaceHolder = () => (
+    <LoadingPlaceholder width={CONTRIBUTE_CARD_WIDTH} heigth={CONTRIBUTE_CARD_HEIGHT} />
+  );
+  const dynamicOptions = { loading: StyledDragDropPlaceHolder, ssr: true };
+  const StyledDragDrop = dynamic(
+    () => import(/* webpackChunkName: 'StyledDragDrop' */ '../StyledDragDrop'),
+    dynamicOptions,
+  );
+
+  const getOtherWaysToContributeOrder = memoizeOne(() => {
```

Comment:
Using `memoizeOne` in the render func is a bad practice because the function will be reset at each render